src/go/k8s/pkg/resources/statefulset.go

```diff
@@ -269,6 +285,10 @@ func (r *StatefulSetResource) obj(
 		externalAddressType = externalListener.External.PreferredAddressType
 	}
 	tlsVolumes, tlsVolumeMounts := r.volumeProvider.Volumes()
+
+	// We set statefulset replicas via status.replicas in order to control it from the handleScaling function
+	replicas := r.pandaCluster.Status.GetReplicas()
```

Comment:
Nit: I think normally the operator will update the desired state of StatefulSet then this will reconcile back to the owner which is the Cluster.

Reply:
Yeah, it's a bit unconventional. The problem is that I needed a field in the status to tell all individual reconcilers how to behave. E.g. the current replicas changes the way we populate both statefulset and configmap, but also how we contact the cluster via admin API etc.

So I needed a field to send an input to the other controllers, containing the decision on how many instances to run. I took a shortcut and used `status.replicas` instead of something like `status.desiredReplicas`, since they would have matched at next reconcile loop..

Reply:
yeah maybe we should rename to avoid confusion, I agree the convention is to use replicas a bit differently and I was also confused